python/tvm/target/target.py

```diff
@@ -46,7 +46,7 @@ class Target(Object):
     - :py:func:`tvm.target.intel_graphics` create Intel Graphics target
     """
 
-    def __init__(self, tag_or_str_or_dict):
+    def __init__(self, tag_or_str_or_dict, host_tag_or_str_or_dict=None):
```

Comment:
for the API, I would almost expect `Target(target_host, sub_target_0, sub_target_1, ...)`

what would be the future path towards supporting multiple sub-targets? would there be a target_host object which contains all the sub-targets?

Reply:
In terms of sub-targets, I think it will be better to converge to composite targets instead, where we can specify a list of targets as `--devices`: https://github.com/apache/tvm/blob/main/src/target/target_kind.cc#L311-L313.

We can help create a short cut to construct the composite kind, like adding a static function

```python
@staticmethod
def composite(target, devices):
```

Reply:
> for the API, I would almost expect `Target(target_host, sub_target_0, sub_target_1, ...)`

In terms of API, that would be very good indeed!

If internally we would convert that to represent the required `composite target`, with specific dictionary representing all the internal data structures, etc., that's implementation details the _we care about_, but the end-user, interested in compiling a model shouldn't.

Reply:
@areusch as a follow-up to this, we've recently updated `tvmc` to accept something similar - in terms of user facing syntax to express "targets" for compilation - to what you present here.

Check it out on https://github.com/apache/tvm/pull/7304 for code and https://discuss.tvm.apache.org/t/byoc-partitioning-support-on-tvmc/8901 for discussion.